assets/src/scripts/components/hydrograph/domain.js

```diff
@@ -112,29 +111,108 @@ export const getLowestAbsoluteValueOfTickValues = function(tickValues) {
+        }
+    });
+
+    // remove values that are duplicates
+    roundedTickValues = Array.from(new Set(roundedTickValues));
+
+    return roundedTickValues;
+};
+
+
+/**
+ * Helper function that checks where the x-axis intersects the y-axis and then keeps any additional tick marks with
+ * values above the point of intersection.
+ * @param {array} additionalTickValues, numerical values for y-axis ticks
+ * @param {array} yDomain, an array of two points, the lower and upper extent of the y-axis
+ * @returns {array} tickValuesAboveXaxis, numerical values for y-axis ticks with any value below the y-axis removed
+ */
+export const removeTickValuesBelowXaxis = function(additionalTickValues, yDomain) {
+    let tickValuesAboveXaxis = [];
+
+    for (let value of additionalTickValues) {
+        if (value > yDomain[0]) {
+           tickValuesAboveXaxis.push(value)
+        }
+    }
+
+    return tickValuesAboveXaxis;
+};
+
+
+/**
+ *  Helper function that, when negative values are present on the y-axis, adds additional negative values to the set of
+ *  tick values used to fill tick mark value gaps on the y-axis on some log scale graphs
+ * @param {array} additionalTickValues, a set of tick mark values
+ * @returns {array} additionalTickValues, a set of tick mark values with (when needed) additional negative values
+ */
+export const generateNegativeTicks = function(tickValues, additionalTickValues) {
     if (tickValues.some(value => value < 0)) {
         let tickValueArrayWithNegatives = additionalTickValues.map(x => x * -1);
         additionalTickValues = tickValueArrayWithNegatives.concat(additionalTickValues);
     }
 
-    // round the values to a chosen multiple of a number such as 5
-    additionalTickValues = additionalTickValues.map(value => Math.ceil(value/MULTIPLE_TO_ROUND_TO)*MULTIPLE_TO_ROUND_TO);
+    return additionalTickValues;
+};
+
+
+/**
+ * Function creates a new set of tick values that will fill in gaps in log scale ticks, then combines this new set with the
+ * original set of tick marks.
+ * @param {array} tickValues, the list of y-axis tick values
+ * @returns {array} fullArrayOfTickMarks, the new full array of tick marks for log scales
+ */
+export const getArrayOfAdditionalTickMarks = function(tickValues, yDomain) {
+    let additionalTickValues = [];
+    // get the lowest value of generated tick mark set to use as a starting point for the additional ticks
+    let lowestTickValueOfLogScale = getLowestAbsoluteValueOfTickValues(tickValues);
+
+    // Make a set of tick values that when graphed will have equal spacing on the y axis
+    additionalTickValues = generateAdditionalTickValues(lowestTickValueOfLogScale);
+
+    // round the values to a chosen multiple of a number
+    additionalTickValues = getRoundedTickValues(additionalTickValues);
+
+    // remove any tick values that would be below the y-axis
+    additionalTickValues = removeTickValuesBelowXaxis(additionalTickValues, yDomain);
```

Comment:
Could you instead pass `yDomain` into `getRoundedTickValues`, and just skip over values outside the domain (so `removeTickValuesBelowXaxis` becomes unnecessary)?